File: orgo_mcp/tools/access.py
"""Access tools — VNC password for direct connection."""

import json

from orgo_mcp.server import mcp
from orgo_mcp.auth import get_current_api_key, resolve_computer_id
from orgo_mcp.client import api_request
from orgo_mcp.errors import handle_orgo_error
from orgo_mcp.models import ComputerIdInput


@mcp.tool(
    name="orgo_vnc_password",
    annotations={"readOnlyHint": True, "destructiveHint": False, "idempotentHint": True, "openWorldHint": False},
)
async def orgo_vnc_password(params: ComputerIdInput) -> str:
    """Get the VNC password for direct connection to a computer's display.
    Used for VNC clients, terminal WebSocket, and the orgo-vnc React component."""
    try:
        api_key = get_current_api_key(mcp)
        computer_id = resolve_computer_id(params.computer_id)
        data = await api_request("GET", f"computers/{computer_id}/vnc-password", api_key)
        return json.dumps(data, indent=2)
    except Exception as e:
        return handle_orgo_error(e)

File: src/tools/access.ts
/**
 * Access tools -- VNC password for direct connection.
 */

import { z } from "zod";
import type { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
import { getApiKey, resolveComputerId } from "../auth.js";
import { apiRequest } from "../client.js";
import { handleError } from "../errors.js";

export function registerAccessTools(server: McpServer): void {
  server.tool(
    "orgo_vnc_password",
    "Get the VNC password for direct connection to a computer's display. Used for VNC clients, terminal WebSocket, and the orgo-vnc React component.",
    {
      computer_id: z.string().optional().describe("Computer ID (uses ORGO_DEFAULT_COMPUTER_ID if omitted)"),
    },
    async ({ computer_id }) => {
      try {
        const apiKey = getApiKey();
        const id = resolveComputerId(computer_id);
        const data = await apiRequest("GET", `computers/${id}/vnc-password`, apiKey);
        return { content: [{ type: "text" as const, text: JSON.stringify(data, null, 2) }] };
      } catch (e) {
        return { content: [{ type: "text" as const, text: handleError(e) }], isError: true };
      }
    }
  );
}
